site/snippets/web3auth.ts

```diff
@@ -1,29 +1,21 @@
-import { WalletClientSigner, type SmartAccountSigner } from "@alchemy/aa-core";
-import { Web3Auth } from "@web3auth/modal";
-import { createWalletClient, custom } from "viem";
+import { Web3AuthSigner } from "@alchemy/aa-signers";
 
-// see https://web3auth.io/docs/quick-start for more info
-const web3auth = new Web3Auth({
-  // web3auth config...
-});
+export const createWeb3AuthSigner = async () => {
+  const web3AuthSigner = new Web3AuthSigner({
+    clientId: "test",
+    chainConfig: {
+      chainNamespace: "eip155",
+    },
+  });
 
-await web3auth.initModal();
+  await web3AuthSigner.authenticate({
+    init: async () => {
+      await web3AuthSigner.inner.initModal();
+    },
+    connect: async () => {
+      await web3AuthSigner.inner.connect();
+    },
```

Comment:
if we require `modal` to be installed can we provide these as the default params for `authenticate`?